aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1477 @@
+        // flow.
+        if (!coin::is_account_registered<AptosCoin>(address_of(&multisig_signer))) {
+            coin::register<AptosCoin>(&multisig_signer);
+        };
+
+        (multisig_signer, multisig_signer_cap)
+    }
+
+    fun create_multisig_account_seed(seed: vector<u8>): vector<u8> {
+        // Generate a seed that will be used to create the resource account that hosts the staking contract.
+        let multisig_account_seed = vector::empty<u8>();
+        vector::append(&mut multisig_account_seed, DOMAIN_SEPARATOR);
+        vector::append(&mut multisig_account_seed, seed);
+
+        multisig_account_seed
+    }
+
+    fun validate_owners(owners: &vector<address>, multisig_account: address) {
+        let distinct_owners = simple_map::create<address, bool>();
+        let i = 0;
+        let len = vector::length(owners);
+        while (i < len) {
+            let owner = *vector::borrow(owners, i);
+            assert!(owner != multisig_account, error::invalid_argument(EOWNER_CANNOT_BE_MULTISIG_ACCOUNT_ITSELF));
+            assert!(
+                !simple_map::contains_key(&distinct_owners, &owner),
+                error::invalid_argument(EDUPLICATE_OWNER),
+            );
+            simple_map::add(&mut distinct_owners, owner, true);
+            i = i + 1;
+        }
+    }
+
+    fun assert_is_owner(owner: &signer, multisig_account: &MultisigAccount) {
+        assert!(
+            vector::contains(&multisig_account.owners, &address_of(owner)),
+            error::permission_denied(ENOT_OWNER),
+        );
+    }
+
+    fun num_approvals_and_rejections(owners: &vector<address>, transaction: &MultisigTransaction): (u64, u64) {
+        let num_approvals = 0;
+        let num_rejections = 0;
+
+        let votes = &transaction.votes;
+        let len = vector::length(owners);
+        let i = 0;
+        while (i < len) {
+            let owner = *vector::borrow(owners, i);
+            i = i + 1;
+
+            if (!simple_map::contains_key(votes, &owner)) {
+                continue
+            } else if (*simple_map::borrow(votes, &owner)) {
+                num_approvals = num_approvals + 1;
+            } else {
+                num_rejections = num_rejections + 1;
+            };
+        };
+
+        (num_approvals, num_rejections)
+    }
+
+    fun assert_multisig_account_exists(multisig_account: address) {
+        assert!(exists<MultisigAccount>(multisig_account), error::invalid_state(EACCOUNT_NOT_MULTISIG));
+    }
+
+    fun add_multisig_account_references(
+        owners: &vector<address>, multisig_account: address) acquires OwnedMultisigAccounts {
+        let len = vector::length(owners);
+        let i = 0;
+        while (i < len) {
+            let owner = *vector::borrow(owners, i);
+            if (!exists<OwnedMultisigAccounts>(owner)) {
+                move_to(&create_signer(owner), OwnedMultisigAccounts {
```

Comment:
this use of `create_signer` is also not very .. idiomatic

Reply:
Yeah but at the same time, we're starting to explore allowing creating known resources in any accounts. The implicit registration flow is one example where users can create the CoinStore<CoinType> resource in any arbitrary accounts. We should make sure our entire stack is equipped to handle large amounts of resources. May be this would be lessened if object is widely adopted, but we'll see.